master/internal/task/postgres_allocations.go

```diff
@@ -34,3 +34,67 @@ WHERE allocation_id = ?
 
 	return &allocation, nil
 }
+
+// InsertTrialAllocationWorkspaceRecord inserts a record linking an trial's allocation
+// to a trial to it's respective workspace & experiment.
+func InsertTrialAllocationWorkspaceRecord(
+	ctx context.Context,
+	experimentID int,
+	allocationID model.AllocationID,
+) error {
+	workspaceInfo := model.Workspace{}
+	err := db.Bun().NewSelect().Model(&workspaceInfo).
+		Join("INNER JOIN projects p on workspace.id = p.workspace_id").
+		Join("INNER JOIN experiments e on p.id = e.project_id").
+		Where("e.id = ?", experimentID).Scan(ctx)
+	if err != nil {
+		return fmt.Errorf(
+			"unable to retrieve workspace information for allocation (%s) associated with experiment %d: %w",
+			allocationID,
+			experimentID,
+			err,
+		)
+	}
+
+	_, err = db.Bun().NewInsert().Model(&model.AllocationWorkspaceRecord{
+		AllocationID:  allocationID,
+		ExperimentID:  experimentID,
+		WorkspaceID:   workspaceInfo.ID,
+		WorkspaceName: workspaceInfo.Name,
+	}).Exec(ctx)
+	if err != nil {
+		return fmt.Errorf("inserting allocation workspace record: %w", err)
+	}
+	return nil
+}
+
+// InsertNTSCAllocationWorkspaceRecord inserts a record linking
+// an NTSC tasks' allocation to it's respective workspace.
+func InsertNTSCAllocationWorkspaceRecord(
```

Comment:
I think these functions could use a couple unit tests -- if this is really urgent for release party, a nice compromise would be to create/link a ticket in the backlog for this. The RM team has an epic on increasing test coverage for Bun funcs.

Reply:
This is a good call out! I went ahead and added some Postgres integration tests on these functions to make sure that the expected information was persisted